uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -812,6 +814,72 @@
         });
       });
 
+      describe('action matrix for a leave request', function () {
+        var actionMatrix;
+
+        describe('status: awaiting approval', function () {
+          beforeEach(function () {
+            actionMatrix = getActionMatrixForStatus('waiting_approval');
+          });
+
+          it('shows the "respond" and "cancel" actions', function () {
+            expect(actionMatrix).toEqual(['respond', 'cancel']);
+          });
+        });
+
+        describe('status: more information required', function () {
+          beforeEach(function () {
+            actionMatrix = getActionMatrixForStatus('more_information_requested');
+          });
+
+          it('shows the "edit" and "cancel" actions', function () {
+            expect(actionMatrix).toEqual(['edit', 'cancel']);
+          });
+        });
+
+        describe('status: approved', function () {
+          beforeEach(function () {
+            actionMatrix = getActionMatrixForStatus('approved');
+          });
+
+          it('shows the "edit" action', function () {
+            expect(actionMatrix).toEqual(['edit']);
+          });
+        });
+
+        describe('status: cancelled', function () {
+          beforeEach(function () {
+            actionMatrix = getActionMatrixForStatus('cancelled');
+          });
+
+          it('shows the "edit" action', function () {
+            expect(actionMatrix).toEqual(['edit']);
+          });
+        });
+
+        describe('status: rejected', function () {
+          beforeEach(function () {
+            actionMatrix = getActionMatrixForStatus('rejected');
+          });
+
+          it('shows the "edit" action', function () {
+            expect(actionMatrix).toEqual(['edit']);
+          });
+        });
+
+        /**
+         * Calls the controller method that returns the action matrix for
+         * a given Leave Request in a particular status
+         *
+         * @param  {string} statusName
+         * @return {Array}
+         */
+        function getActionMatrixForStatus(statusName) {
```

Comment:
Should not this helper method be placed outside in the same level of `compileComponent `?